packages/templates/src/cli.js

```diff
@@ -0,0 +1,149 @@
+  await page.waitForSelector('div.templateGridItem');
+
+  // total templates to generate screenshots from
+  const templateCount = await page.evaluate(() => {
+    return document.querySelectorAll('.templateGridItem').length;
+  });
+
+  // Now let's go through each template
+  for (
+    let currentTemplate = 1;
+    currentTemplate <= templateCount;
+    currentTemplate++
+  ) {
+    // Get the template name and remove spaces to use as part of file name.
+    const templateName = await page.evaluate((gridItemId) => {
+      return document
+        .querySelector(
+          `div#template-grid-item-${gridItemId} div[data-testid="card-action-container"]`
+        )
+        .getAttribute('aria-label')
+        .replace(/\s/g, '');
+    }, currentTemplate);
+
+    console.log(`getting screenshots for ${templateName}`);
+
+    // Use the template, this will open up the editor
+    await page.click(
+      `div#template-grid-item-${currentTemplate} button[aria-label="Use this template"]`
+    );
+
+    // Load the template in the editor to get story id to render preview
+    await page.waitForNavigation({ waitUntil: 'networkidle2' });
+    const editorUrl = await page.url(); // 'http://localhost:8899/wp-admin/post.php?post=1163&action=edit#page=%252217fa9c85-89c7-47d0-a7a8-29f6ef56e161%2522'
+    const storyId = editorUrl.split('post=')[1].split('&')[0];
+
+    const previewUrl = `http://localhost:8899/?post_type=web-story&p=${storyId}&preview=true`;
+
+    // Need to click preview for the story to save and be loadable in a second window.
+    await page.waitForSelector('aria/Preview');
+    await page.click('aria/Preview');
+
+    const pagePreview = await browser.newPage();
+    await pagePreview.setViewport({
+      width: 1600,
+      height: 800,
+      deviceScaleFactor: 2,
+    });
+    // set prefers-reduced-motion to get story without animations so screenshots are complete page views
+    await pagePreview.emulateMediaFeatures([
+      { name: 'prefers-reduced-motion', value: 'reduce' },
+    ]);
+    await pagePreview.evaluate(
+      () => matchMedia('(prefers-reduced-motion: reduce)').matches
+    );
+
+    await pagePreview.goto(previewUrl);
+    // Ensure that the page's ready before looping
+    await pagePreview.waitForSelector('aria/Next page');
+
+    const totalPages = await pagePreview.evaluate(() => {
+      return document.querySelectorAll('amp-story-page').length;
+    });
+
+    for (let currentPage = 1; currentPage < totalPages; currentPage++) {
+      const templatePageSafeArea = await pagePreview.$(
+        `amp-story-page:nth-child(${currentPage}) .page-safe-area`
+      );
+      await templatePageSafeArea.screenshot({
+        path: `${screenshotsPath}/${templateName}-${currentPage}.png`,
+      });
+
+      if (currentPage !== totalPages - 1) {
+        await pagePreview.click('aria/Next page');
+        // Wait 2 seconds, prevents screenshots not capturing the full page. Without this some pages won't load in time.
```

Comment:
What exactly are we waiting for here? Network requests?

Reply:
Fly in animations!

Reply:
Fly in and pan and zoom animations!
I'll run a sample without this delay to show you - it seems related to pan and zoom or fly in. 

Even with the reduce motion browser setting fly in elements were being captured at their start size and the screenshots were just really sad.

Reply:
Doesn't the `prefers-reduced-motion` trick stop animations from happening?

If not, perhaps we can speed up animations or disable them using CSS.

A quick search reveals https://github.com/puppeteer/puppeteer/issues/453#issuecomment-324310130 and https://github.com/puppeteer/puppeteer/issues/511 and https://stackoverflow.com/questions/51547994/wait-for-transition-to-end-in-puppeteer

Reply:
i'll test a few things out

Reply:
alright, so i was wrong about it being based on a few animations. It seems to just be limitations of emulating `prefers-reduced-motion` because when I also set my system preferences to be reduced motion I got perfect snapshots.  I'm thinking maybe I was seeing the carousel page motion so the page was just being captured as a smaller portion of the screenshot? 

Thanks for the links to other people having similar issues. 

[Pausing all animations for consistent state](https://github.com/puppeteer/puppeteer/issues/511):  This thread is interesting, what I thought would get us closest is style injections - there's a screenshot below of that. Basically looks the same as emulating reduced motion.

[Waiting for transitions to end: ](https://stackoverflow.com/questions/51547994/wait-for-transition-to-end-in-puppeteer) : This is really tricky because we don't know what all transitions we're waiting on, it's any assortment of elements. Seems cumbersome. 

[Playback rate:](https://github.com/puppeteer/puppeteer/issues/453#issuecomment-324310130) This might work in tandem with the reduce motion emulation. 

For comparison (note, i ditched the timeout completely for all of these):

| Reduce Motion Sys Prefs Only | Emulate Reduce Motion Only | No Reduce Motion | Both Reduce Motion Sys Prefs & Emulate |
| --- | --- | --- | --- |
| ![3-no-timeout-motion-reduced-machine-no-emmulation](https://user-images.githubusercontent.com/10720454/129776189-e2ecaddd-42a4-4e44-9843-c67e53a26281.png) |  ![3-no-timeout-no-reduce-motion](https://user-images.githubusercontent.com/10720454/129776945-426629ec-25fc-4434-a6d0-d62bfa73a2b0.png) | ![3-no-timeout](https://user-images.githubusercontent.com/10720454/129776302-0b54ffd9-4687-453d-87fd-431765b3c14c.png) | ![3-no-timeout-both-reduce](https://user-images.githubusercontent.com/10720454/129776511-24cfb2e5-8a42-4d3d-89ab-db947c7b1e55.png) |


|  Style Injection & Emulate Reduce Motion | Style Injection | Playback Rate Sped Up | Playback Rate Sped Up With Emulate Reduce Motion | 
 | --- | --- | --- | --- |
| ![3-emulate-reduce-motion-style-injection](https://user-images.githubusercontent.com/10720454/129777793-ba0d2218-a79d-4895-ab77-b6636aeb18ca.png) | ![3-style-injection](https://user-images.githubusercontent.com/10720454/129777839-284f604f-e4f5-4fea-aa9e-4d50103ae5e5.png) | ![3-playbackrate2-no-emulate](https://user-images.githubusercontent.com/10720454/129779138-7f0fcc5c-7a6d-4fed-b492-5151d399e26d.png) | ![3-playbackrate2](https://user-images.githubusercontent.com/10720454/129778987-141db9e6-2ddd-4503-ae60-befa8aed596a.png) | 


I think I'm going to run with speeding up the playback and emulating so that future people running this script don't have to set their system preferences before running the script. I'll see if I run into more issues - this Ace hotel template was the real offender though from last week.

Reply:
Thanks for this thorough analysis! 😻 

A bit suprised that `Emulate Reduce Motion Only` doesn't have the same effect as `Reduce Motion Sys Prefs Only` 🤷 

Your proposed plan makes sense 👍

Reply:
It's odd, isn't it? 

It feels like it also might be some race conditions happening, while speeding up and emulating helps it doesn't solve everything when I run the loop of 52 templates. I had to rerun a few individual templates to fix some screenshots - they would be fine on some runs and not the others. In any case, it feels like solving it entirely is a time suck. With this method we'll atleast be sure that getting proper screenshots of the pages with all their elements exposed is possible, it just might need to be run a second time on a single template.